src/panda_eup/pbd_interface.py

```diff
@@ -35,95 +36,117 @@ def __init__(self):
             else:
                 self.default_parameters[parameter_name] = rospy.get_param('~' + parameter_name,
                                                                           self.default_parameters[parameter_name])
-
-        self.gripper_state_subscriber = rospy.Subscriber("/franka_gripper/joint_states", JointState,
-                                                         self.gripper_state_callback)
-
-        try:
-            self.kinesthetic_client = rospy.ServiceProxy('/primitive_interface_node/kinesthetic_teaching',
-                                                         EnableTeaching)
-        except rospy.ServiceException:
-            rospy.logerr('Cannot create Kinesthetic Teaching client!')
-
-        try:
-            self.kinesthetic_client.wait_for_service(5.0)
-        except rospy.ROSException:
-            rospy.logerr('Cannot contact the Primitive Interface Node!')
-
-        self.interpreter = interpreter.PandaProgramInterpreter()  # internal interpreter, for execute_now_primitives
-
+        if not self.robotless_debug:
+            self.gripper_state_subscriber = rospy.Subscriber("/franka_gripper/joint_states", JointState,
+                                                            self.gripper_state_callback)
+
+            try:
+                self.kinesthetic_client = rospy.ServiceProxy('/primitive_interface_node/kinesthetic_teaching',
+                                                            EnableTeaching)
+            except rospy.ServiceException:
+                rospy.logerr('Cannot create Kinesthetic Teaching client!')
+
+            try:
+                self.kinesthetic_client.wait_for_service(5.0)
+            except rospy.ROSException:
+                rospy.logerr('Cannot contact the Primitive Interface Node!')
+
+        #The interpreter of panda_widgets is used for the tuning and running of existing programs, the interpreter of pbd_interface
```

Comment:
Please turn this in a block comment, as it is too long. Also, lines should be have <120 characters (PEP8)

```
# The interpreter of panda_widgets is used for the tuning and running of existing programs
# The interpreter of pbd_interface is used for actions that are targeted at the program-to-be-created,
# such as moving to previous preconditions when primitives are deleted
```